libraries/chain/controller.cpp

```diff
@@ -4128,11 +4130,19 @@ struct controller_impl {
       EOS_ASSERT( id == bsp->id(), block_validate_exception,
                   "provided id ${id} does not match block id ${bid}", ("id", id)("bid", bsp->id()) );
 
+
       if constexpr (is_proper_savanna_block) {
+         assert((qc && verify_qc_future.valid()) || (!qc && !verify_qc_future.valid()));
```

Comment:
or 
```suggestion
         assert(!!qc == verify_qc_future.valid());
```

Reply:
Hmm, ok. Not sure that is more readable, but sure.